src/coord/src/catalog/migrate.rs

```diff
@@ -34,116 +39,122 @@ pub const CONTENT_MIGRATIONS: &[fn(&mut Catalog) -> Result<(), anyhow::Error>] =
     // will fail.
     //
     // Introduced for v0.6.1
-    |catalog: &mut Catalog| {
-        struct TypeNormalizer;
-
-        impl<'ast> VisitMut<'ast, Raw> for TypeNormalizer {
-            fn visit_data_type_mut(&mut self, data_type: &'ast mut DataType<Raw>) {
-                if let DataType::Other { name, .. } = data_type {
-                    let mut unresolved_name = name.name().clone();
-                    if unresolved_name.0.len() == 1 {
-                        unresolved_name = UnresolvedObjectName(vec![
-                            Ident::new(PG_CATALOG_SCHEMA),
-                            unresolved_name.0.remove(0),
-                        ]);
-                    }
-                    *name = match name {
-                        RawName::Name(_) => RawName::Name(unresolved_name),
-                        RawName::Id(id, _) => RawName::Id(id.clone(), unresolved_name),
+    Migration {
+        name: "rewrite-type-references",
```

Comment:
I think these names should be unambiguous. e.g. We're going to potentially rewrite type references again. Maybe append the version number onto them?

Reply:
The error message will always include the version number as well, does that help? I'd rather not require a human ensure the correct count.

Reply:
I think Sean's point is that someone will want to come along and add a new copy of `rewrite-type-references` so the name might as well be "rewrite-type-references-<MATERIALIZE VERSION INTRODUCED>`.

Since the comment format says "introduced for ...", maybe just delete those comments and instead encode them in a `introduced_in_version: &str` field on the `Migration` struct?

This is all distinct from the catalog content "version", which is just the migration index.

Reply:
Ah, like the materialized version? Yeah I can do that, it might be potentially annoying because of our releases not always being the next release (e.g. 0.8.4 is going to be 0.9.0) but that's probably fine.